src/app.rs

```diff
@@ -217,63 +241,27 @@ impl<'a, T, E: Send + Sync + Clone + 'static> Application<'a, T, E> {
     }
 
     /// Advances the game world by one tick.
-    fn advance_frame(&mut self) {
+    fn advance_frame(&mut self)
+    where
+        for<'b> R: EventReader<'b, Event = E>,
+    {
         trace!("Advancing frame (`Application::advance_frame`)");
 
         {
-            let world = &mut self.world;
-            let states = &mut self.states;
             #[cfg(feature = "profiler")]
             profile_scope!("handle_event");
 
-            let window_events = world
-                .read_resource::<EventChannel<Event>>()
-                .read(&mut self.window_reader_id)
-                .cloned()
-                .map(|e| StateEvent::Window(e))
-                .collect::<Vec<_>>();
-
-            for event in window_events {
-                if !self.ignore_window_close {
-                    if cfg!(target_os = "ios") {
-                        if let &StateEvent::Window(Event::WindowEvent {
-                            event: WindowEvent::Destroyed,
-                            ..
-                        }) = &event
-                        {
-                            states.stop(StateData::new(world, &mut self.data));
-                        }
-                    } else {
-                        if let &StateEvent::Window(Event::WindowEvent {
```

Comment:
This needs to stay. Optimally, you could toggle this behavior through a bool that would default to true.

Reply:
Fixed.